examples/azure/terraform-azure-nsg-example/README.md

```diff
@@ -0,0 +1,129 @@
+1. Run `terraform init`.
+1. Run `terraform apply`.
+1. When you're done, run `terraform destroy`.
+
+## Running automated tests against this module
+
+1. Sign up for [Azure](https://azure.microsoft.com/).
+1. Configure your Azure credentials using one of the [supported methods for Azure CLI
+   tools](https://docs.microsoft.com/en-us/cli/azure/azure-cli-configuration?view=azure-cli-latest).
+1. Install [Terraform](https://www.terraform.io/) and make sure it's on your `PATH`.
+1. [Review environment variables](#review-environment-variables).
+1. Install [Golang](https://golang.org/) and make sure this code is checked out into your `GOPATH`.
+1. `cd test`
+1. Make sure [the azure-sdk-for-go versions match](#check-go-dependencies) in [/test/go.mod](/test/go.mod) and in [test/azure/terraform_azure_nsg_example_test.go](/test/terraform_azure_nsg_example_test.go).
+1. `go build terraform_azure_nsg_example_test.go`
+1. `go test -v -run TestTerraformAzureNsgExample`
+
+## Module test APIs
+
+This module exposes the following top-level API methods:
+
+- GetDefaultNsgRulesClientE returns a rules client that retuns the _default_ NSG rules<br>
+`func GetDefaultNsgRulesClientE(subscriptionID string) (network.DefaultSecurityRulesClient, error)`
+- GetCustomNsgRulesClientE returns a rules client that returns the _custom_ NSG rules<br>
+`func GetCustomNsgRulesClientE(subscriptionID string) (network.SecurityRulesClient, error)`
+- GetAllNSGRulesE returns a slice containing all rules from the NSG (including defauls and custom)<br>
+`func GetAllNSGRulesE(resourceGroupName, nsgName, subscriptionID string) (NsgRuleSummaryList, error)`
+
+Note that the `GetAllNsgRulesE` method returns an `NsgRuleSummaryList`, which is defined as follows:
+
+- NsgRuleSummaryList holds a colleciton of NsgRuleSummary structs<br>
+` type NsgRuleSummaryList struct { SummarizedRules []NsgRuleSummary } `
+
+This collection wrapper has the following method:
+
+- FindRuleByName looks for a matching rule name <br>
+`func (summarizedRules *NsgRuleSummaryList) FindRuleByName(t *testing.T, name string) NsgRuleSummary`
+
+Finally, the inner collection is defined as an `NsgRuleSummary` struct, defined as follows:
+
+```
+type NsgRuleSummary struct {
+	Name                     string
+	Description              string
+	Protocol                 string
+	SourcePortRange          string
+	DestinationPortRange     string
+	SourceAddressPrefix      string
+	DestinationAddressPrefix string
+	Access                   string
+	Priority                 int32
+	Direction                string
+}
+```
+
+This wrapper has the following methods available:
+
+- AllowsDestinationPort checks to see if the rule allows a specific destination port<br>
+`func (summarizedRule *NsgRuleSummary) AllowsDestinationPort(port string) bool`
+- AllowsSourcePort checks to see if the rule allows a specific source port<br>
+`func (summarizedRule *NsgRuleSummary) AllowsSourcePort(port string) bool`
+
+## Check Go Dependencies
```

Comment:
Same question as in  https://github.com/gruntwork-io/terratest/pull/626/files#r490158408 as to why this is necessary?

Reply:
Azure API's do not offer a simple way to determine if rules are enabling or blocking traffic. These helper methods have proven quite useful to make sure the network spec is working as-defined.